pkg/xds/topology/dataplanes_test.go

```diff
@@ -91,7 +93,7 @@ var _ = Describe("Resolve Dataplane address", func() {
 			}
 			actual := topology.ResolveAddresses(core.Log, lif, given)
 			Expect(actual).To(HaveLen(3))
-			Expect(actual).To(Equal(expected))
+			Expect(expected).Should(BeComparableTo(expected, cmp.Comparer(proto.Equal)))
```

Comment:
If these are protobufs you want to compare there's a `matchers.ProtoMatch` in pkg/test/matchers

Reply:
See: https://github.com/kumahq/kuma/pull/4742

Reply:
The issue is that the value is not a `proto.Message` but it contains nested `proto.Messages`